vm_supervisor/vm/firecracker/instance.py

```diff
@@ -115,5 +120,89 @@ async def wait_for_init(self) -> None:
 
     async def configure(self):
         """Configure the VM by sending configuration info to it's init"""
-        # TODO: Implement Cloud-init interface
         pass
+
+    async def _create_user_data_file(self) -> str:
+        """Creates user data configuration file for cloud-init tool"""
+
+        ssh_authorized_keys = self.resources.message_content.authorized_keys \
+            if self.resources.message_content.authorized_keys \
+            else []
+
+        config = {
+            "hostname": self.vm_hash,
+            "disable_root": False,
+            "ssh_pwauth": True,
+            "ssh_authorized_keys": ssh_authorized_keys,
+        }
+
+        cloud_config_header = "#cloud-config\n".encode()
+        config_output = yaml.safe_dump(config, default_flow_style=False, sort_keys=False).encode()
+
+        with NamedTemporaryFile(delete=False) as yaml_file:
+            yaml_file.write(cloud_config_header)
+            yaml_file.write(config_output)
+            yaml_file.flush()
+
+        return yaml_file.name
+
+    async def _create_network_file(self) -> str:
+        """Creates network configuration file for cloud-init tool"""
+
+        ip = self.tap_interface.guest_ip.with_prefixlen
+        route = str(self.tap_interface.host_ip).split("/", 1)[0]
+
+        network = {
+            "network": {
+                "ethernets": {
+                    "eth0": {
+                        "dhcp4": False,
+                        "dhcp6": False,
+                        "addresses": [
+                            str(ip)
+                        ],
+                        "gateway4": route,
+                        "nameservers": {
+                            "addresses": [
+                                "8.8.8.8",
+                                "8.8.4.4",
+                            ],
+                        },
+                    },
+                },
+                "version": 2,
+            },
+        }
+
+        network_output = yaml.safe_dump(network, default_flow_style=False, sort_keys=False).encode()
+
+        with NamedTemporaryFile(delete=False) as network_yaml_file:
+            network_yaml_file.write(network_output)
+            network_yaml_file.flush()
+
+        return network_yaml_file.name
+
+    async def _create_cloud_init_drive(self) -> Drive:
+        """Creates the cloud-init volume to configure and setup the VM"""
+
+        assert (
+                self.enable_networking and self.tap_interface
+        ), f"Network not enabled for VM {self.vm_id}"
+
+        disk_image_path = Path(f"/opt/cloud-init-{self.vm_hash}.img")
```

Comment:
All data on disk must be stored in either `/var/lib/aleph` or `/var/cache/aleph`.